### PR TITLE
Application tracking

### PR DESCRIPTION
**Pull request by hx-markterry**
# What does this change?

Start of explaining how tracking should now be working.
# Why?

To avoid confusion for developers as more tracking is implemented.
# Main reviewers:
- [x] @hpoom 
- [x] @viktort 

Ps.
  Bonus feature; I've updated the table of contents to be more readable and include pages that others have forgotten to add.

### CODE REVIEW

**Comment by hpoom** [comment]
@panicnot would be good to get your eyes on this please.

**Comment by hpoom** [comment]
I like the use of the word `Application` but also wonder if we should change it to `site` to make these guidelines more inclusive of non-booking flow areas, like the landing sites (also know as SEO sites). Sentences like `Applications should have tracking enabled` do not read right for those areas, unless we mean  Application as in HX SEO, not as in the Site www.holidayextras.co.uk

File: application-tracking.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Applications should have tracking enabled for all features that a user can interact with. When new features are written adding tracking should be a requirement for the definition of done.
+
+In addition to the previous point extra explicit tracking should also be performed when additional information needs to be gathered. Examples of these are as follows:
+
+ * Adding an item to a basket - Product info required.
+ * Booking success - Product, upgrade and booking info required.
+ 
+### How?
+
+At an application level we should not be concerned with how data is tracked beyond calling our tracking module with the correct parameters, once this has been done we can trust that the tracking module will do its job. Currently we have more than one tracking system and have plans to change how these are implemented if at all. Abstracting the decision away from an application level allows changes to our current systems and implementing new systems to be much easier.
```
Comment (hpoom) [inline application-tracking.md]: `Currently we have more than one tracking system and have plans to change how these are implemented if at all.` does not read well. Try...

`Currently we have more than one tracking system and have plans to change how these are implemented. We also plan to deprecate some of tracking solutions.`

**Comment by hx-markterry** [comment]
@hpoom I'm using `application` rather than `site` as I we may be tracking stuff from a server or app too.

File: application-tracking.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+## Principles
+
+### Why?
+
+We should not be asking why we should be tracking an event in an application, we should be tracking events by default. Events containing sensitive information such as user or card details should still be tracked but with the sensitive data removed or masked.
+
+### What?
+
+Applications should have tracking enabled for all features that a user can interact with. When new features are written adding tracking should be a requirement for the definition of done.
```
Comment (panicnot) [inline application-tracking.md:20]: it is possible to also track what a customer sees but doesn't interact with, can be handy for marketing banners etc

**Comment by hpoom** [comment]
I have :+1: it is a good starting doc. I do expect to see more PR's in time from all to keep expanding this doc.

File: application-tracking.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Application Tracking
+
+## Principles
+
+### Why?
+
```
Comment (chrisken) [inline application-tracking.md:6]: I thought the "why" could do with some high level context.

Tracking is really about communication - it is a protocol for the developers, via the data team, to tell the business how their changes work (and in the middle here is being able to read data in an automated and meaningful way). 

Tracking is not just about "web tracking" but covers multiple mechanisms of performance monitoring.  

Can I suggest the following:

We build applications to support the business and drive revenue. 

To associate the performance of applications and the behaviour of customers with commercial performance and revenue, we instrument our applications with _Event Tracking_.

_Events_ form sequences of activity (clicked X, viewed Y), which can be correlatated with outcomes (made a booking, added an upgrade).
Event sequences can be separated into different streams, or segments (by specific user, by product, by version of the application).  This sequences form the foundation of business monitoring.

**Tracking is a fundamental communication mechanism between application development and commercial performance.**

Events can come from many sources, not just explicit web tracking but also application logfiles, operational databases or external systems, such as Responsys.

File: application-tracking.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+## Current goal
+
+* One single tracking module to provide an agnostic API for all our application tracking needs.
+* All our applications / modules that require tracking will use the same tracking module.
+* Our current tracking systems will sit behind this thin layer.
+* Decide on tracking event data schema, so we are not bound to one or more current systems.
+* Keep specific application conventions and / or technology out of the tracking module.
+* Tracked elements are not accessed directly by the tracking module.
+* Each tracking event will be sent to all current systems (actual data may vary, base on system).
```
Comment (viktort) [inline application-tracking.md]: typo - base -> based

File: application-tracking.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+* Different applications have varying levels of tracking.
+* Different areas of the business prefer to use different systems for reporting.
+* Tracking data is created via code and by accessing an element's HTML5 data attributes.
+
+## Current goal
+
+* One single tracking module to provide an agnostic API for all our application tracking needs.
+* All our applications / modules that require tracking will use the same tracking module.
+* Our current tracking systems will sit behind this thin layer.
+* Decide on tracking event data schema, so we are not bound to one or more current systems.
```
Comment (viktort) [inline application-tracking.md:52]: ... data team/engineers to be part of any discussions and should be aware of any schema changes in the future before they are implemented.

**Comment by viktort** [comment]
Agree, this is a good start of the culture doc - will :+1: once @chrisken 's comments and mine are addressed.

On another note - are we going to have links pointing to private google docs for application documentation? Thinking about the best way to let people know where these resources are internally but try and keep it as centralised as possible.

File: application-tracking.md (hunk 3 above)
Comment (hx-markterry) [inline application-tracking.md:6]: Section now expanded.

**Comment by hx-markterry** [comment]
@hpoom @panicnot @viktort @chrisken I've addressed your comments on this PR, could you take another look please, thanks.

**Comment by viktort** [comment]
:+1:

**Comment by hpoom** [comment]
:+1:  from me

**Pull request merged**